.github/workflows/nightly-release.yml

```diff
@@ -14,12 +14,12 @@ jobs:
           should-continue: ${{ steps.generate.outputs.should-continue }}
           version: ${{ steps.generate.outputs.version }}
           release-id: ${{ steps.generate.outputs.release-id }}
-        permissions: write-all # might not be used for realy deployment
+        permissions: write-all
         steps:
           - name: Checkout Code
             uses: actions/checkout@v4
             with:
-              ref: 'main'
+              ref: 'main' # only work on 'main' branch
```

Comment:
@JasonYangShadow I can't decide whether I think this needs to specify `fetch-depth: 0` to allow `git describe` to generate versions based on the recent tag. Right now, this is generating releases like `warewulf-6a9e161.tar.gz`; but with tags it would be `warewulf-4.5.0rc3+113-g6a9e1616.tar.gz`. For most releases I think using the tag names is better; but I'm not certain for the nightly.

What do you think?

Reply:
See https://github.com/warewulf/warewulf/pull/1197 for reference.

Reply:
yeah, I think we can change to use the tag you mentioned above. Just need to retrieve the version info from the new `get-verison.sh` I guess.